packages/forms/src/Components/Concerns/EntanglesStateWithSingularRelationship.php

```diff
@@ -203,7 +203,17 @@ public function getCachedExistingRecord(): ?Model
             return $this->cachedExistingRecord;
         }
 
-        $record = $this->getRelationship()?->getResults();
+        $relationshipName = $this->getRelationshipName();
+
+        if (! $relationshipName) {
+            return null;
+        }
+
+        if ($this->getModelInstance()->relationLoaded($relationshipName)) {
+            $record = $this->getModelInstance()->getRelationValue($relationshipName);
```

Comment:
I think we should first save `$parentRecord` as `$this->getRecord()`, early return null if that is null, then use that instead of calling `$this->getModelInstance()` twice?

Reply:
it's done.